### DISCUSSION TITLE
Loading OSM map failed to terriamap catalog

### DISCUSSION THREAD

**Discussion by eliadd** [TerriaJS/terriajs#7239] [Q&A]
Hi,

I am tring to load OSM 3d layer from the following link https://www.arcgis.com/home/item.html?id=ca0470dbbddb4db28bad74ed39949e25 (https://basemaps3d.arcgis.com/arcgis/rest/services/OpenStreetMap3D_Buildings_v1/SceneServer).
With the following configuration on the Catalog (TerriaMap)):
{
      "id": "OSMBuildingsTrees",
      "name": "OpenStreetMap 3D Buildings & Trees",
      "type": "open-street-map",
      "url": "https://basemaps3d.arcgis.com/arcgis/rest/services/OpenStreetMap3D_Buildings_v1/SceneServer?layers=ca0470dbbddb4db28bad74ed39949e25",
      "description": "3D buildings and trees from OpenStreetMap provided by ArcGIS."
    }

The OSM layer displayed on the workbanch but not displayed on the earth map.

If you can please elaborate what I am doing wrong.

Thx,

Eliad.

**Comment by ljowen** [comment]
Hi @eliadd 

It looks like there's an issue with our underlying rendering engine with this layer, I've reproduced it here: (Zoom in to see the console error) [Sandcastle link](https://sandcastle.cesium.com/#c=jVcLcxu3Ef4rWzYT303lO0VSGtsi2dq07GoiW2qotNMJMzZ4AElEIHAD4EjTDv97d3E48o5iHh6NpQN2F/v49lugMNp5WEmxFhYGoMUaRsLJapn9J6wlk14RvkdGeya1sJPeCXyZaAAvrMWVF43Cff2dzaxZ/tdYxeNCkp6QONNyybw0qDBjyomw6OVSKLS6X9umlxM90XkO74wVwJlnTngH3uBPZaFgGqYCGOeCw0JYkWVZEL9fCKisgpI5h1sof30+fo3qd9asJMfwXFWWxqItZRiXeg4MHP5SAq41F59Q6fw1jAuhBSSom4JiG1RLsv7COD/Mw9ZY2JWwedhyeV/yYQrGoqnCKCUKChDMDFwwU0sdtZAC5Qn1WmsYSBHqQZE6LEfI86T3HqvyP2MfJr0XtEBLC+9L9yLPp5idJSvdOc+YLebSZYVZ5vWfuRXO5w5NS6xhflsKPfZWCP+OleevP7yqpKI0uA+rb9qe/aP2elCw04vvTvl0yvn0gk/Pnk0Z/+5C8PPnzy+ei7NvEQhYrlAtbze1r1iHay29ZEp+xiI1GIGyqcF6IYtF8+lgLozkmDqN5zpK3VT4tRAa5patpN+AFYp5rEziN6UsmFIbKusOFinCioNQSpYOLTEFlBAeXVkIOV94cBvnxRLLUAMVAjghojPNGr9n4BfSAf5og0qlKORMCn5C5fUIrh9/uKFNqVcYHUehx96vpVIBnWWpULUxTdB0WNKixjNUBFDMSjjPW6ZdYQIgY61gzRyaXZZSoWDASVNvwaxfPJV6ZjI9Z9lSqvyvnk0/rOfu2cXTYJ0ygnZDIgCdunr77uz09Bm8jSl9Z7hQ5FmNtRDEOB48ALZm0jct/dIWb6/H9+TGlRKr0L4xb01XhXb/0arkEJketVwblGEh/3zm9fWDvfn32asfPtu/3x3HavQ5v16yeYTlpEcnEDuEnI6sQGCAPHdgSvIrUAQ1f/xmRAWWLYWnFsSUzyoFM6wlF9NqPg/9j5laSVcRWvF7nxO0ehuNxiaEOk8hFQcZeNFJ4Qk5F9wgzNQIiWGRnbpf2hE0cN71SMeNHXkdVOaA2w6qEPjjpzZx/HxS7+wja7IZvXnJefCYElrz3q5nEYzknVxiY68ihLEdQzz15MgC3WU7Gaw750nL/dZBI6GxIsjjS2EZ4ZMSYDSCT/onNTdTAyO1oIxHskgw3Qb58kuTGKNEpsw8xvrxnqZAaBkcDUEWCkotVZgiIvuUK9gX7KsvQW77sUnCNs6cl2hBe+oSM/0F2TzyVejrhVHUtTMDbGoqH2wXlbWogKzkBLE/tSseXaFDBDWSnppPwKUrMakNvTeyV/VRnbFbryXkFPozXpg1IIRZSHqgDJpWM/PKfEJ22WUfWV2PS1aIqxW68y8EtqJ14a91WfmXYSwls0rH+aRvxMyPlCweQhKXZiWWqDfR6Y7G73ATy06exYj2uEQmfhD8TQx0cAACstpYzErjJJ1Z962cQfKXGCgXMxz8POkYS9Om3azAbx20tjV22mffRbPHDm/2jjvRuFGf8nu+IGkhUrVP4euv/7x0RjSXIfTfI82mAV9NSLumpi30/I/1L2PTor/duHc2obGXUeO8kUJxl6QZwqRV76QlDqCERw5EzMgyphCJAkHtvZXTygs36V12pTX2akxd/a+OZBZOQ/3GhbnwtQdvjL3r+JvsdeGghPudtHUshXwbWjB7EBuX1GelmRJ67hcwhNNOTNCNqL0B8KSPExJveYVCWh40F9owRLGNnmI5WaX8PclMesO+nxq+GT65bBuhZk5aUb/HjFA7Rre6rtTeN0mpRbIFc7drmhmlsNjfOyvpI+VuLH8bwMe+t+jWYvjVl53atp/jQt/z4cfLP1KPau6nnfbPpM6H+J89on+sSwkC1Ah4syADfy4OCMhBgDw6/9GZ2+5C53P7W3UOucnreuFvKmAnnG0XT0djOnA7OjzpVfpBm7Xu9ELLXpfCs6gWG+U3hLpN1/pqqTRsdjghugs7hW3TM9tIlFuc8zHM8cFUuMeRnd1cvbn/MLq5Hn1fk+GOIw7qWsY/Qmn3o9c3CWrvZ97cGPR6hDfOZE/0tWMkj9yEV3cxwhstswKflLQ46aUwGAzgFH799XdEcQ53pDuMWg8JvEJXopkUu8XwoKTRDr2TXt/5DYKjSds/5ZJegvReTLIsx9dBSc8Ml08rpCafFc41ie3nbdU+lyuQfHDkVbyjF7xnqjE+f4hLcpR/pBqfn7d4qaVLQW/nVn/xzfCm3sRXbT/Hz8aJY3a8MWrK7ME5/wc)

I'll report the issue on the Cesium community forum

**Reply by ljowen** [reply]
I believe I've found the cause within Cesium. issue created here: https://github.com/CesiumGS/cesium/issues/12114

**Reply by Techmaven-geospatial** [reply]
The easiest solution is to use tile server https://tileserver.techmaven.net
It includes in its 3D map Terriajs available to any one in that network along with a highly advanced OpenLayers map.
It does all the data serving including 
Basemaps, overlays, quantize mesh terrain tiles 
3dtiles, 3d models 
And also includes searchable API OGC API FEATURES 

Can also access terriajs on ios, android and windows via https://EarthExplorer.techmaven.net that also works offline disconnected

**Reply by eliadd** [reply]
Thx @Techmaven-geospatial  - It is a win desktop solution (https://tileserver.techmaven.net/) - can you recommand on a tile server that can be deplyed as docker/linux machine?

**Comment by eliadd** [comment]
Thx @ljowen !!!

I have another Q that relates to this issue - in my POC of the TerriaMap product, I need to run it on a stand-alone machine with no internet connetion.
I triyed to simulate it on my development enviroment & disable the internet connection - the globe map was not loaded + all links for the baseMap return with errors of no resource found (because I disabled the internet connection) + google fonts that were not found.
So my Q - how can I run the TerriaMap solution on isolated machine with no internet connection?

**Comment by Techmaven-geospatial** [comment]
Tile server has been updated to the latest.net Maui and supports fully cross platform all the versions and distros of Linux and Mac 

It's also in a ready to go Rocky Linux OVA OVF
OPEN VIRTUALIZATION FORMAT

**Reply by eliadd** [reply]
Thx @Techmaven-geospatial  - also, I asked it in this thread below - how to handle all the css calls to google-api from the TerriaMap server in stand alone deployment (with no internet)?

**Comment by Techmaven-geospatial** [comment]
You'll have to clarify I'm not familiar with any Google API calls 

We're not using that for any search or any base Maps where is Google being used other than for photo realistic 3D buildings?

**Reply by eliadd** [reply]
I am asking regarding the TeeriaMap server - in the build folder under the wwwroot we have the index.TerriaMaps.css file with the following internet referance:
@import url(https://fonts.googleapis.com/css?family=Nunito:300,400,500,400italic,600,700&display=swap);
@import url(https://fonts.googleapis.com/css?family=Nunito:600,800&display=swap);

**Comment by Techmaven-geospatial** [comment]
Our version of Terriajs for edge web or part of tile server and geodata server and portable tile server have no CDN or Internet hosted content

This is used in heavy production like us Forest Service in Alaska or on military bases around the world by different armed forces or by humanitarian organizations. 
As well as part of command and control common operating picture Solutions

**Reply by eliadd** [reply]
Will it work with TerriaMap server as it is with the regular distribution of Terriajs?
Can you please point me to the url where I can dowonlad it - the 'Terriajs for edge web' you mention
